packages/react-native-codegen/src/parsers/flow/modules/index.js

```diff
@@ -181,7 +181,17 @@ function translateTypeAnnotation(
           return emitRootTag(nullable);
         }
         case 'Promise': {
-          return emitPromise(hasteModuleName, typeAnnotation, parser, nullable);
+          return emitPromise(
+            hasteModuleName,
+            typeAnnotation,
+            parser,
+            nullable,
+            types,
+            aliasMap,
+            tryParse,
+            cxxOnly,
+            translateTypeAnnotation,
```

Comment:
Here I would call the `translateTypeAnnotation` to get the `elementType` to promote simpler functions and composability. What do you think?

Reply:
answered above